scene/gui/dialogs.cpp

```diff
@@ -299,10 +343,20 @@ AcceptDialog::AcceptDialog() {
 	set_transient(true);
 	set_exclusive(true);
 	set_clamp_to_embedder(true);
+	set_flag(FLAG_BORDERLESS, true);
+	// set_flag(FLAG_RESIZE_DISABLED, true);
 
 	bg = memnew(Panel);
 	add_child(bg);
 
+	title = memnew(Label);
+	title->set_align(Label::ALIGN_CENTER);
+	title->set_anchor(MARGIN_RIGHT, Control::ANCHOR_END);
+	add_child(title);
+	close_btn = memnew(TextureButton);
```

Comment:
Window close/maximize/minimize buttons probably should be controlled and drawn be DisplayServer instead of using normal controls.

For example on macOS window buttons are on the left, and maximize/zoom button provides menu with Split View controls, which is impossible to reproduce, but OS provides API for placing these buttons anywhere in the window. Custom controls will not have full functionality, and buttons on the right will be out of place.

Reply:
Since this is probably not required for windows it can check for a DisplayServer feature and use it if available, like:

```c++
if (DispalyServer::get_singleton()->has_feature(FEATURE_CUSTOM_WINDOW_BUTTONS)) {
    DispalyServer::get_singleton()->window_set_custom_buttons(custom_tilte_bar_rect, window_id);
} else {
    // use custom controls
}
```

Reply:
Indeed DisplayServer is a more elegant approach to this (how did I not think of that XD).
But I'm not sure if win32 provides an API to move the title bar buttons: if the DisplayServer approach is used, we have to  either keep the native buttons or use a set of custom buttons in Windows only (possibly X11 as well).

Reply:
Actually, it's probably even better to use normal (non-borderless) window and extend context to the title, in this case there's no need to configure individual button and window border but it's possible to draw title using custom theme:

```obj-c++
		[wd.window_object setTitlebarAppearsTransparent:YES];
		[wd.window_object setStyleMask:[wd.window_object styleMask] | NSWindowStyleMaskFullSizeContentView];
```

<img width="288" alt="Screenshot 2020-07-16 at 13 37 27" src="https://user-images.githubusercontent.com/7645683/87661654-a4499280-c769-11ea-9d4d-9db30d47e9a0.png">